warzone2100/riscv64.patch

```diff
@@ -15,3 +13,6 @@ index 1d7846c..4ecf847 100644
      -DCMAKE_INSTALL_PREFIX=/usr \
      -DCMAKE_INSTALL_LIBDIR=lib \
      -DWZ_DISTRIBUTOR="ArchLinux"
++  sed -i 's$-lpthread$-pthread$g' build.ninja
```

Comment:
(Currently) CMake appears to ignore `-DCMAKE_THREAD_PREFER_PTHREAD=TRUE`, and modifying CMakeLists.txt doesn't help. After some attempts (passing it via command line, modifying CMakeLists.txt and src/CMakeLists.txt), CMake insisted on generating `-lpthread` instead of `-pthread` into build.ninja.


```patch
--- /src/CMakeLists.txt
+++ /src/CMakeLists.txt
@@ -803,3 +803,8 @@
                message( WARNING "Unable to get OUTPUT_NAME from warzone2100 target" )
        endif()
 endif()
+
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
+target_link_libraries(warzone2100 Threads::Threads)
```

The patch above couldn't persuade CMake to use `-pthread`, and CMakeLists.txt in the root directory haven't been changed since 4 Oct 2021.

Therefore, I choose to modify `build.ninja` directly because CMake seems to change its behavior on `pthread`.

Reply:
This looks like a cmake issue then. I've heard about glibc 2.34+ stopping cmake from using `-pthread` before but I don't have details for that specific issue atm.